lib/src/models/chat_params.dart

```diff
@@ -27,12 +27,29 @@ class ChatParams {
           'maxTokens must be greater or equal to 0',
         );
 
+  /// the name of the model to chat with, e.g. mistral-tiny
   final String model;
+
+  /// an array of messages to chat with, e.g.
+  /// `{role: 'user', content: 'What is the best French cheese?'}`
   final List<ChatMessage> messages;
+
+  /// the temperature to use for sampling, e.g. 0.5.
   final double? temperature;
+
+  /// the cumulative probability of tokens to generate, e.g. 0.9.
+  /// Defaults to null.
   final double? topP;
+
+  /// the maximum number of tokens to generate, e.g. 100. Defaults
+  /// to null.
   final int? maxTokens;
+
+  /// whether to use safe prompt, e.g. true. Defaults to False.
   final bool? safePrompt;
+
+  /// the random seed to use for sampling, e.g.
+  /// 42. Defaults to null.
```

Comment:
I would personally put the `Defaults to XYZ` in separate lines, but that shouldn't block the PR 😉

Reply:
For now, I would keep it as is. If we put it in a separate line then it will be hidden in the generated dart docs (the user will have to go to this field's subpage in docs to see the full description).